packages/drippie-mon/src/service.ts

```diff
@@ -0,0 +1,117 @@
+import { BaseServiceV2, Counter, validators } from '@eth-optimism/common-ts'
+import { Provider } from '@ethersproject/abstract-provider'
+import { ethers } from 'ethers'
+import * as DrippieArtifact from '@eth-optimism/contracts-periphery/artifacts/contracts/universal/drippie/Drippie.sol/Drippie.json'
+
+type DrippieMonOptions = {
+  rpc: Provider
+  drippieAddress: string
+}
+
+type DrippieMonMetrics = {
+  executableDripCount: Counter
+}
+
+type DrippieMonState = {
+  drippie: ethers.Contract
+}
+
+export class DrippieMonService extends BaseServiceV2<
+  DrippieMonOptions,
+  DrippieMonMetrics,
+  DrippieMonState
+> {
+  constructor(options?: Partial<DrippieMonOptions>) {
+    super({
+      name: 'drippie-mon',
+      loop: true,
+      loopIntervalMs: 60_000,
+      options,
+      optionsSpec: {
+        rpc: {
+          validator: validators.provider,
+          desc: 'Provider for network where Drippie is deployed',
+        },
+        drippieAddress: {
+          validator: validators.str,
+          desc: 'Address of Drippie contract',
+        },
+      },
+      metricsSpec: {
+        executableDripCount: {
+          type: Counter,
+          desc: 'Number of times a drip has spotted being executable',
+          labels: ['name'],
+        },
+      },
+    })
+  }
+
+  protected async init(): Promise<void> {
+    this.state.drippie = new ethers.Contract(
+      this.options.drippieAddress,
+      DrippieArtifact.abi,
+      this.options.rpc
+    )
+  }
+
+  protected async main(): Promise<void> {
+    const dripCreatedEvents = await this.state.drippie.queryFilter(
```

Comment:
Does this default to the range [start, end] or just tip? Will fail on backends like alchemy if range is too large

Reply:
Actually should work fine on Alchemy, which is why I did it this way. Alchemy is [10k blocks OR 2k events] I believe. Since we don't expect to have many drips, should work fine to search the entire chain because Alchemy caches these results.